src/Core/System/Snippet/Command/Util/CountryAgnosticFileLinter.php

```diff
@@ -0,0 +1,216 @@
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            if (!$isAdminTranslationFile && !$isCoreTranslationFile) {
+                continue;
+            }
+
+            $currentFileData = $isAdminTranslationFile ? $adminFileData : $coreFileData;
+
+            $formatedLanguageStruct = $this->createTranslationFile(
+                $currentFileData,
+                $file,
+                (bool) $isAdminTranslationFile,
+                $countrySpecificFileCollection,
+            );
+
+            $languageFiles->add($formatedLanguageStruct);
+        }
+
+        return $this->processAgnosticFiles(new LintedTranslationFileStruct(
+            $languageFiles,
+            $countrySpecificFileCollection,
+        ));
+    }
+
+    public function fixFilenames(LintedTranslationFileStruct $lintedFileStruct): void
+    {
+        foreach ($lintedFileStruct->getFixingCollection() as $translationFile) {
+            $this->filesystem->rename(
+                $translationFile->getFullPath(),
+                $translationFile->getAgnosticPath(),
+            );
+        }
+    }
+
+    private function processAgnosticFiles(LintedTranslationFileStruct $lintedFileStruct): LintedTranslationFileStruct
+    {
+        $specificCollection = $lintedFileStruct->getSpecificCollection();
+        if ($specificCollection->count() === 0) {
+            return $lintedFileStruct;
+        }
+
+        $translationCollection = $lintedFileStruct->getCompleteCollection();
+        foreach ($specificCollection as $countrySpecificFile) {
+            // If no agnostic file exists, $countrySpecificFile is content for `fixFilenames` to be fixed
+            if ($translationCollection->get($countrySpecificFile->getAgnosticPath()) === null) {
+                $lintedFileStruct->addFixableFile($countrySpecificFile);
+            }
+        }
+
+        return $lintedFileStruct;
+    }
+
+    private function getFinder(LintedTranslationFileOptions $options): Finder
+    {
+        $finder = (new Finder())
+            ->files()
+            ->ignoreUnreadableDirs()
+            ->ignoreDotFiles(true)
+            ->ignoreVCS(true)
+            ->exclude([
+                'node_modules',
+                'vendor',
+                'bin',
+                'static',
+                // Translations of languages fetched from crowdin should not be linted
+                'SwagLanguagePack/src/Resources/snippet',
+                'SwagLanguagePack/src/Resources/app/administration/src/snippet',
+                ...$options->ignoredPaths,
+            ])
+            ->name([SnippetPatterns::CORE_SNIPPET_FILE_PATTERN, SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN])
+            ->sortByName(true);
+
+        if ($options->dir) {
+            $finder->in($options->dir);
+        } elseif (empty($options->extensionPaths)) {
+            $finder->in('src');
+
+            if ($options->isAll) {
+                $finder->in('custom/plugins');
```

Comment:
not sure, if we should include "vendor" here, or if the "all" option could be achieved in another way :thinking: